File: python/31.py
#!/usr/bin/env python3

import sys

coins = [1,2,5,10,20,50,100,200]

def get_ways(val, current):
    if val <= 0:
        return 1
    else:
        s = 0
        for c in coins:
            if val >= c and c >= current:
                s += get_ways(val - c, c)
        return s

print(get_ways(int(sys.argv[1]), 0))

File: rust/solutions/31.rs
use std::env;

static COINS: &'static [i32] = &[1, 2, 5, 10, 20, 50, 100, 200];

fn get_ways(val: i32, current: i32) -> i32 {
    if val <= 0 {
        return 1;
    }
    else {
        let mut sum = 0;
        for c in COINS {
            if val >= *c && *c >= current {
                sum += get_ways(val - *c, *c);
            }
        }
        return sum;
    }
}

fn main() {
    let args: Vec<String> = env::args().collect();
    let val: i32 = args[1].parse().unwrap();
    println!("{}", get_ways(val, 0));
}
